src/Sensor.cc

```diff
@@ -196,6 +213,55 @@ bool SensorPrivate::SetTopic(const std::string &_topic)
   return true;
 }
 
+//////////////////////////////////////////////////
+void Sensor::PublishMetrics(const ignition::common::Time &_now)
+{
+  return this->dataPtr->PublishMetrics(_now);
+}
+
+void SensorPrivate::PublishMetrics(const ignition::common::Time &_now) {
+  if(!this->performanceSensorMetricsPub)
+  {
+    this->performanceSensorMetricsPub =
+      node.Advertise<msgs::PerformanceSensorMetrics>(
+          this->topic + "/performance_metrics");
+  }
+  if (!performanceSensorMetricsPub ||
+      !performanceSensorMetricsPub.HasConnections())
+  {
+    return;
+  }
+
+  // Computes simulation update rate and real update rate.
+  double simUpdateRate;
+  double realUpdateRate;
+  const auto clockNow = std::chrono::steady_clock::now();
+  if(this->lastUpdateTime > 0)
+  {
+    const double diffSimUpdate = _now.Double() -
+      this->lastUpdateTime.Double();
+    simUpdateRate = 1.0 / diffSimUpdate;
```

Comment:
Just to be safe, I think we should check if the diffs are not zero before dividing

Reply:
Nice catch.
Done.